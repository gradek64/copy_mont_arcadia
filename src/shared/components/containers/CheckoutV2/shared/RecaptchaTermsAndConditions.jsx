import React from 'react'
import PropTypes from 'prop-types'

const RecaptchaTermsAndConditions = (props, { l }) => {
  const url = 'https://policies.google.com'
  const renderLink = (text, linkName) =>
    `<a
          rel="noopener noreferrer"
          target="_blank"
          href=${`${url}/${linkName}`}
        >
          ${text}
        </a>`
  const privacyLink = renderLink(l`Recaptcha Privacy Policy`, 'privacy')
  const termsLink = renderLink(l`Recaptcha Terms of Service`, 'terms')
  const content = l`This site is protected by reCAPTCHA and the Google ${privacyLink} and ${termsLink} apply.`

  return (
    <div
      className="TermsAndConditions" // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: content }}
    />
  )
}

RecaptchaTermsAndConditions.contextTypes = {
  l: PropTypes.func,
}

export default RecaptchaTermsAndConditions
