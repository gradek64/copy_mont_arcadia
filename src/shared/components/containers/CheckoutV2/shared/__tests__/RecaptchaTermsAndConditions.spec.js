import testComponentHelper from 'test/unit/helpers/test-component'
import RecaptchaTermsAndConditions from '../RecaptchaTermsAndConditions'

describe('<RecaptchaTermsAndConditions />', () => {
  const renderComponent = testComponentHelper(RecaptchaTermsAndConditions)

  describe('@renders', () => {
    it('should render default state', () => {
      const { getTree } = renderComponent()
      expect(getTree()).toMatchSnapshot()
    })

    it('should render a div with className equal to `TermsAndConditions`', () => {
      const { wrapper } = renderComponent()
      expect(wrapper.find('.TermsAndConditions')).toHaveLength(1)
    })
  })
})
