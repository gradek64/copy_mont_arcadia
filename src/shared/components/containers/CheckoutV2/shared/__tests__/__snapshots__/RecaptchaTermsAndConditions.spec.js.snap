// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecaptchaTermsAndConditions /> @renders should render default state 1`] = `
<div
  className="TermsAndConditions"
  dangerouslySetInnerHTML={
    Object {
      "__html": "This site is protected by reCAPTCHA and the Google <a
          rel=\\"noopener noreferrer\\"
          target=\\"_blank\\"
          href=https://policies.google.com/privacy
        >
          Recaptcha Privacy Policy
        </a> and <a
          rel=\\"noopener noreferrer\\"
          target=\\"_blank\\"
          href=https://policies.google.com/terms
        >
          Recaptcha Terms of Service
        </a> apply.",
    }
  }
/>
`;
